infrastructure/provider/flight_reserve.go

```diff
@@ -0,0 +1,42 @@
+package provider
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"github.com/cyneptic/letsgo/internal/core/entities"
+)
+
+type SortFilterProviderClient struct {
+	client  *http.Client
+	address string
+}
+
+func NewSortFilterProviderClient(address string) *SortFilterProviderClient {
+	tr := &http.Transport{}
+	cl := &http.Client{Transport: tr}
+
+	return &SortFilterProviderClient{
+		client:  cl,
+		address: address,
```

Comment:
make a const on top of this file and include the address there, change the function to not take an address from another location, this should separate concerns and we can modify only the provider file in order to use another provider.